crox/src/main.rs

```diff
@@ -163,3 +159,31 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     Ok(())
 }
+
+fn timestamp_to_min_max(timestamp: Timestamp) -> (SystemTime, SystemTime) {
+    match timestamp {
+        Timestamp::Instant(t) => (t, t),
+        Timestamp::Interval { start, end } => {
+            // Usually start should always be greater than end, but let's not
+            // choke on invalid data here.
+            (cmp::min(start, end), cmp::max(start, end))
+        }
+    }
+}
+
+fn make_start_timestamp(data: &ProfilingData) -> SystemTime {
+    // We cannot assume the first event in the stream actually is the first
+    // event because interval events are emitted at their end. So in theory it
+    // is possible that the event with the earliest starting time is the last
+    // event in the stream (i.e. if there is an interval event that spans the
+    // entire execution of the profile).
+    //
+    // Let's be on the save side and iterate the whole stream.
+    let min = data
+        .iter()
+        .map(|e| timestamp_to_min_max(e.timestamp).0)
+        .min()
+        .unwrap();
+
+    min - Duration::from_micros(1)
+}
```

Comment:
I think this shall be moved to profiling_data.rs and in the future use the start_time from the metadata header

Reply:
Seems like a good idea.

Reply:
I'll leave a comment. I don't want to do it as part of this PR as it would require adding metadata decoding too and I want to keep this a refactoring PR as much as possible.